lib/streamlit/type_util.py

```diff
@@ -445,11 +453,16 @@ def convert_anything_to_df(df: Any) -> DataFrame:
     if is_type(df, "numpy.ndarray") and len(df.shape) == 0:
         return pd.DataFrame([])
 
-    if is_type(df, _SNOWPARK_DF_TYPE_STR) and not isinstance(df, list):
-        df = pd.DataFrame(df.take(MAX_UNEVALUATED_DF_ROWS))
-        if df.shape[0] == MAX_UNEVALUATED_DF_ROWS:
+    if (
+        is_type(df, _SNOWPARK_DF_TYPE_STR)
+        and not isinstance(df, list)
```

Comment:
nit: I think the `and not isinstance(df, list)` is not really necessary here, since if it is of type `_SNOWPARK_DF_TYPE_STR` or `_SNOWPARK_TABLE_TYPE_STR` it anyways cannot be a list. An alternative might be something like:

```python
is_snowpark_dataframe(df) and not isinstance(df, list)
```

Reply:
It's updated now 👍